hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/BaseCompiler.groovy

```diff
@@ -108,6 +110,8 @@ abstract class BaseCompiler extends BaseVisitor implements HadoopDslCompiler {
       cleanBuildDirectory(file);
     }
 
+    doAtCompilerCreation();
```

Comment:
Can we name this function more appropriately? The current name sounds like this function should be called in the constructor of each compiler.